x-pack/plugins/timelines/server/search_strategy/timeline/index.ts

```diff
@@ -5,48 +5,166 @@
 import {
   TimelineFactoryQueryTypes,
   TimelineStrategyResponseType,
   TimelineStrategyRequestType,
+  EntityType,
 } from '../../../common/search_strategy/timeline';
 import { timelineFactory } from './factory';
 import { TimelineFactory } from './factory/types';
+import {
+  ENHANCED_ES_SEARCH_STRATEGY,
+  ISearchOptions,
+} from '../../../../../../src/plugins/data/common';
 
 export const timelineSearchStrategyProvider = <T extends TimelineFactoryQueryTypes>(
-  data: PluginStart
+  data: PluginStart,
+  alerting: AlertPluginStartContract
 ): ISearchStrategy<TimelineStrategyRequestType<T>, TimelineStrategyResponseType<T>> => {
+  const esAsInternal = data.search.searchAsInternalUser;
   const es = data.search.getSearchStrategy(ENHANCED_ES_SEARCH_STRATEGY);
+
   return {
     search: (request, options, deps) => {
-      if (request.factoryQueryType == null) {
+      const factoryQueryType = request.factoryQueryType;
+      const entityType = request.entityType;
+      const alertConsumers = request.alertConsumers;
+
+      if (factoryQueryType == null) {
         throw new Error('factoryQueryType is required');
       }
-      const queryFactory: TimelineFactory<T> = timelineFactory[request.factoryQueryType];
-      const dsl = queryFactory.buildDsl(request);
-      return es.search({ ...request, params: dsl }, options, deps).pipe(
-        map((response) => {
-          return {
-            ...response,
-            ...{
-              rawResponse: shimHitsTotal(response.rawResponse, options),
-            },
-          };
-        }),
-        mergeMap((esSearchRes) => queryFactory.parse(request, esSearchRes))
-      );
+
+      const queryFactory: TimelineFactory<T> = timelineFactory[factoryQueryType];
+
+      if (entityType != null && entityType === EntityType.ALERTS) {
+        return timelineAlertsSearchStrategy({
+          es: esAsInternal,
+          request,
+          options,
+          deps,
+          queryFactory,
+          alerting,
+          alertConsumers: alertConsumers ?? [],
+        });
+      } else {
+        return timelineSearchStrategy({ es, request, options, deps, queryFactory });
+      }
     },
     cancel: async (id, options, deps) => {
       if (es.cancel) {
         return es.cancel(id, options, deps);
+      } else if (esAsInternal.cancel) {
+        return esAsInternal.cancel(id, options, deps);
       }
     },
   };
 };
+
+const timelineSearchStrategy = <T extends TimelineFactoryQueryTypes>({
+  es,
+  request,
+  options,
+  deps,
+  queryFactory,
+}: {
+  es: ISearchStrategy;
+  request: TimelineStrategyRequestType<T>;
+  options: ISearchOptions;
+  deps: SearchStrategyDependencies;
+  queryFactory: TimelineFactory<T>;
+}) => {
+  const dsl = queryFactory.buildDsl(request);
+  return es.search({ ...request, params: dsl }, options, deps).pipe(
+    map((response) => {
+      return {
+        ...response,
+        ...{
+          rawResponse: shimHitsTotal(response.rawResponse, options),
```

Comment:
we do not need spreading here

Reply:
This code wasn't changed from the original (it looks like it's new because it was moved around). But yea, don't see why the second spread is needed.